sourcegraph/sourcegraph
# JetBrains: Add JS→Java bridge

Fixes https://github.com/sourcegraph/sourcegraph/issues/34748 and https://github.com/sourcegraph/sourcegraph/issues/34697.

When checking the code, it probably makes sense to look at the individual commits to understand the changes, but note that the commit "Handle two return arguments right in bridge in case of an error" is twice because I forgot to press "Amend commit". These two should really be one commit, sorry.

## Test plan

- Run the IDE with a light theme and see if the search box uses its light theme.
- Run the IDE with a dark theme and see if the search box uses its dark theme.
- Run the standalone server (`yarn build && yarn standalone`) and see if the search box correctly appears with the dark theme and a red Search button.

Demonstrated here in a 2-min Loom video: https://www.loom.com/share/3af865da0bdd419080c7995c16186bdf

As seen in the video, there are some glitches, but to my best knowledge, none of them are introduced by this change.

## App preview:

- [Web](https://sg-web-dv-jetbrains-add-js-to-java-bridge.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ceavstxyzu.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
